view/readrom.js

```diff
@@ -4,55 +4,74 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-function load_bin(file){
+/**
+ * 
+ * @param {*} dev 
+ * @param {*} file 
+ * @param {*} reader_method 
+ * @param {*} callback 
+ * @param {*} external_params 
+ * You can use variable `load_returns` to get the return value of `dev.loadFile`.
+ */
+function loadToDevice(dev, file, reader_method, loadfile_external_params=[], callback){
     let reader = new FileReader();
-    const isos = $("#os").prop("checked");
     reader.addEventListener('load', function(e) {
         let binary = e.target.result;
-        if (isos) {
-            zealcom.rom.loadFile(binary);
-            $("#binready").addClass("ready");
-        } else {
-            const addr = $("#address").val();
-            const result = parseInt(addr, 16);
-            zealcom.ram.loadFile(result, binary);
-        }
+        let load_returns = dev.loadFile(binary, ...loadfile_external_params);
+        callback(load_returns);
     });
-    if (typeof file !== "undefined") {
-        reader.readAsBinaryString(file);
+    if (file) {
+        reader[reader_method](file);
     }
 }
 
-$("#read-button").on('click', function() {
+function loadRom(file_rom) {
+    /* Read the rom file */
+    if (file_rom) {
+        loadToDevice(zealcom.rom, file_rom, "readAsBinaryString", [], () => {
```

Comment:
I am not found of the function name as string. A cleaner way would be to have `loadToDevice` only initialize the listener and retrun a `FileReader` object. Then you could have the actual read function called here.